test/ServiceBus.Tests/TestStreamProviders/EHStreamProviderWithSlowConsumingPressureDetecting.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Orleans.Providers.Streams.Common;
+using Orleans.Runtime;
+using Orleans.ServiceBus.Providers;
+using Orleans.Streams;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ServiceBus.Tests.TestStreamProviders
+{
+    public class EHStreamProviderWithSlowConsumingPressureDetecting : PersistentStreamProvider<EHStreamProviderWithSlowConsumingPressureDetecting.AdapterFactory>
+    {
+        public class AdapterFactory : EventHubAdapterFactory
+        {
+            public AdapterFactory()
+            {
+                CacheFactory = CreateQueueCache;
+            }
+
+            private IEventHubQueueCache CreateQueueCache(string partition, IStreamQueueCheckpointer<string> checkpointer, Logger log)
+            {
+                var bufferPool = new FixedSizeObjectPool<FixedSizeBuffer>(adapterSettings.CacheSizeMb, () => new FixedSizeBuffer(1 << 20));
```

Comment:
Just to make it blatantly obvious, maybe have a const for this value called `Megabyte` or `BlockSize` with a comment saying it's 1MB